frontend/static/js/services/authenticationService.js

```diff
@@ -1,18 +1,24 @@
 import ARequestService from "./ARequestService.js";
 import backendURL from "../constants.js";
 
+
+// TODO: Extend for tournaments
 export default class AuthenticationService extends ARequestService {
 	constructor() {
 		super();
 	}
 
-	async postRequest({username, password}) {
+	async postMatch({username, password}) {
 		return super.postRequest(
-			`${backendURL.authenticationURL}`,
+			`${backendURL.authenticationURL}match/ `, //${urlParams ? `/${urlParams}` : ''}`,
 			JSON.stringify({
 				username: username,
 				password: password,
 			})
 		);
 	}
+
+	async postAiMatch() {
+		return super.postRequest(`${backendURL.authenticationURL}match/`);
```

Comment:
I am really not sure what I need to do to get a token for the ai. There are the decorators on the view, which force me to user userdata in the body and th ai_opponent is a url parameter ??

Reply:
There is one token that is related to two users, and the db at least is okay the other user being null. If the decorator stops you from doing a POST request without the other user in the request body, than the decorator should be taken out of the server or changed to be okay with not getting tve other user if tje ai_opponent parameter is found. In general, I dont think you need another function to just pass the url oarameter but rather feed it into the service as is, like in FriendService GET all function. Check it out